components/calendar/src/iso.rs

```diff
@@ -458,9 +467,21 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1237-L1258
     pub(crate) fn iso_from_fixed(date: i32) -> Date<Iso> {
         let year = Self::iso_year_from_fixed(date);
-        let prior_days = date - Self::iso_new_year(year);
+        // Increase the date by 365 (see below why) and set it to next_year, only if the current year is MIN_YEAR
+        let (fixed_date, adjusted_year) = if year == MIN_YEAR {
+            // Gets the next possible ISO year
+            let next_year = year.saturating_add(1);
+            // Note: The min year, -5879610, is not a leap year, and neither are either of the adjacent years,
+            // so we add 365 to keep these calculations in range of an i32.
+            let year_length = 365;
+            (date.saturating_add(year_length), next_year)
```

Comment:
Nit (optional): Neither of these need to be saturating anymore, but I think the compiler should figure it out (at least for the first one)